### DISCUSSION TITLE
How to use `useFetch` in a function?

### DISCUSSION THREAD

**Discussion by ikadix** [nuxt/nuxt#17385] [Questions] [answered]
I'd like to group multiple `useFetch` calls within a function that I can call on user actions. 

Previously I could do this, but recently upgrading I am seeing that calling multiple in a function now results in one of the calls failing. 

I have created a small recreation here, Showing that running the same `useFetch ` calls in the base of the script setup, it will retrieve both, but using a function containing the same functionality will fail and retrieve only 1. I am unsure if there is something I am doing wrong here or if I am misunderstanding how to use `useFetch`

https://stackblitz.com/edit/github-mkvnzb?file=components%2Fgrouped.vue

I am seeing this issue when retrieving the second request in the `grouped.vue` file. 

```
Error: nuxt instance unavailable
    at Module.useNuxtApp (file:///home/projects/github-mkvnzb/.nuxt/dist/server/server.mjs:249:13)
    at Module.useAsyncData (file:///home/projects/github-mkvnzb/.nuxt/dist/server/server.mjs:534:38)
    at Module.useFetch (file:///home/projects/github-mkvnzb/.nuxt/dist/server/server.mjs:693:32)
    at getReportData (file:///home/projects/github-mkvnzb/.nuxt/dist/server/server.mjs:2063:68)
    at async setup (file:///home/projects/github-mkvnzb/.nuxt/dist/server/server.mjs:2072:115)
```

**Comment by danielroe** [comment, marked as answer]
You should not `useFetch` in a function. Like all composables, it is meant to be called only in a setup function (or in a composable which is called in a setup function). It can also be called in a Nuxt plugin.

Instead, you should just use `$fetch` if you want to make a network request in a function.

The reason there's a difference between the two examples you quote is that Vue does some special magic to allow for not losing the context between consecutive awaits. But this only works at the top level.

**Reply by tasiotas** [reply]
please add this to documentation. It took me days for figure it out.
